external-crates/move/crates/move-binary-format/src/file_format.rs

```diff
@@ -601,6 +629,117 @@ pub enum Ability {
     Key = 0x8,
 }
 
+#[derive(Clone, Debug, Eq, PartialEq)]
+#[cfg_attr(any(test, feature = "fuzzing"), derive(proptest_derive::Arbitrary))]
+#[cfg_attr(any(test, feature = "fuzzing"), proptest(no_params))]
+#[cfg_attr(feature = "fuzzing", derive(arbitrary::Arbitrary))]
+#[cfg_attr(feature = "wasm", derive(Serialize, Deserialize))]
+pub struct EnumDefinition {
+    /// The `DatatypeHandle` for this `EnumDefinition`. This has the name and the abilities
+    /// for the type.
+    pub enum_handle: DatatypeHandleIndex,
+    /// NB: variant tag == index in this vector
+    pub variants: Vec<VariantDefinition>,
+}
+
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+#[cfg_attr(any(test, feature = "fuzzing"), derive(proptest_derive::Arbitrary))]
+#[cfg_attr(any(test, feature = "fuzzing"), proptest(no_params))]
+#[cfg_attr(feature = "fuzzing", derive(arbitrary::Arbitrary))]
+#[cfg_attr(feature = "wasm", derive(Serialize, Deserialize))]
+pub struct EnumDefInstantiation {
+    /// The `EnumDefinition` for this `EnumDefInstantiation`.
+    pub def: EnumDefinitionIndex,
+    /// The type parameters for this instantiation.
+    pub type_parameters: SignatureIndex,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+#[cfg_attr(any(test, feature = "fuzzing"), derive(proptest_derive::Arbitrary))]
+#[cfg_attr(any(test, feature = "fuzzing"), proptest(no_params))]
+#[cfg_attr(feature = "fuzzing", derive(arbitrary::Arbitrary))]
+#[cfg_attr(feature = "wasm", derive(Serialize, Deserialize))]
+pub struct VariantDefinition {
+    /// The `EnumDefinition` for that this `VariantDefinition` is part of.
+    pub enum_def: EnumDefinitionIndex,
```

Comment:
We can remove this if we want to -- the only nice thing is it gives us an easy way of traversing from "we have a variant in-hand what is the enum that defines it". At the moment I don't think this is readily used anywhere, so again, totally reasonable to remove it, but I'd kinda like to keep it as it might be useful at a later date?

Reply:
It just feels like one extra thing to check, and an extra thing to have to encode. If we need it later, I feel like we should build a map or something to index it, and not force it into the binary format. This isn't a firmly held view but maybe like... a 80% firmly held one :P 
tagging @dariorussi here as well since I assume he will have thoughts